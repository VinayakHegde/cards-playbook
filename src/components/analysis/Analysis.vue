<template>
  <div class="table-analysis" v-if="$store.getters.canAnalyse">
    <h2 class="table-analysis__highcard">
      High Card: <span class="value">{{ $store.getters.highCard }}</span>
    </h2>
    <h2 class="table-analysis__fullhouse">
      Full House Combo:
      <span
        class="value"
        v-if="
          !$store.getters.fullHouseCombi ||
            !$store.getters.fullHouseCombi.length
        "
        >None</span
      >
    </h2>
    <ol class="table-analysis__fullhouse__combis">
      <li
        v-for="set in $store.getters.fullHouseCombi"
        :key="JSON.stringify(set)"
        class="value"
      >
        {{ set.join(", ") }}
      </li>
    </ol>
  </div>
</template>
<style lang="scss" scoped src="./style.scss"></style>
