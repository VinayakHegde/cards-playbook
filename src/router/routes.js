export default [
  {
    path: "/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/new",
    name: "Home",
    meta: {
      header: "CARDS"
    },
    component: () =>
      import(/* webpackChunkName: "home" */ "../views/home/Home.vue")
  },
  {
    path: "/deck/:hash",
    name: "Pack",
    meta: {
      header: "Ordered Pile"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/pack/Pack.vue"),
    props: true
  },
  {
    path: "/error",
    name: "Error",
    meta: {
      header: "404 | Page not found"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/error/Error.vue"),
    props: true
  },
  {
    path: "*",
    redirect: {
      name: "Error"
    }
  }
];
