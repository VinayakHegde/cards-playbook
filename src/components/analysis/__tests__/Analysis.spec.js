import Vuex from "vuex";
import VueRouter from "vue-router";
import { mount, createLocalVue } from "@vue/test-utils";
import storeConfig from "@/store/store-config";
import routes from "@/router/routes";
import Analysis from "../Analysis.vue";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
describe("Analysis.vue", () => {
  it("match the snapshot", () => {
    const store = new Vuex.Store(storeConfig);

    const router = new VueRouter({ routes });
    const wrapper = mount(Analysis, {
      localVue,
      router,
      store
    });
    expect(wrapper).toMatchSnapshot();
  });
});
