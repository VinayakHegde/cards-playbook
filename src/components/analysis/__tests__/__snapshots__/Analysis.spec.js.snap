// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analysis.vue match the snapshot 1`] = ``;
